libs/validation/src/lib/decorators/string/IsStringWithNonzeroLength.ts

```diff
@@ -0,0 +1,17 @@
+import { buildMessage, ValidateBy, ValidationOptions } from 'class-validator';
+
+export const isStringWithNonzeroLength = (test: unknown): test is string =>
+    typeof test === 'string' && test.length > 0;
+
+export function IsStringWithNonzeroLength(validationOptions?: ValidationOptions) {
+    return ValidateBy({
+        name: isStringWithNonzeroLength.name,
+        validator: {
+            validate: (value, _): boolean => isStringWithNonzeroLength(value),
+            defaultMessage: buildMessage(
+                (eachPrefix) => eachPrefix + '$property must be a string with non-zero length',
```

Comment:
Is `$property` a part of class-validator?

Reply:
Yea. Class validator will replace this with the property name.